src/lsdb/loaders/hipscat/read_hipscat.pyi

```diff
@@ -0,0 +1,14 @@
+from typing import overload, Type
```

Comment:
Can we get a module-level docstring to explain why we have this, and how it would be used?